src/components/InRequest/InRequest.tsx

```diff
@@ -155,7 +109,9 @@ export const InForm: FunctionComponent<any> = (props) => {
     data: RadioGroupOnChangeData
   ) => {
     let empType: emptype = data.value as emptype;
-    updateGradeRankOpt(empType);
+    setFormData((f: INewInForm) => {
+      return { ...f, empType: empType as emptype };
+    });
```

Comment:
Understand this alternate way of writing is better suited when using a use state inside a useEffect as it will help prevent dependency issues.  Most of these are going away as I began implementing React-Hook-Form yesterday, so not sure it is worth cleaning them up now.